cmake/build_configurations/compiler_options.cmake

```diff
@@ -29,7 +29,10 @@ SET(CMAKE_CXX_STANDARD 98)
  
 # Compiler options
 IF(UNIX)  
-  SET(CMAKE_CXX98_EXTENSION_COMPILE_OPTION -std=gnu++03)
+  IF(NOT APPLE)
```

Comment:
In theory it's possible to use a recent GCC on macOS, thus I'd try NOT (APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang"). But maybe it's even possible to get a "community" clang version there as well, so maybe it should be a direct CMake compiler runtime check whether the option is accepted. If this is not overengineering.

Reply:
It's also possible with travis, it has a gcc7 profile for example. For this, my starting point was our slack discussion, with the conclusion that we only need clang.

I'll change it to a runtime check then - and this also affects all 3 branches.

Reply:
ack, sorry for not being clear on Slack

Reply:
Updated it, and this also fixed the clang4 issue.